### PR TITLE
maintainability updates

### PR DESCRIPTION
**Pull request by reliq**
Various updates incl:

- group objects by type (for transparency)
- replace `SuffixedNumber` helper with contract-bounded service
- add IDE helper comments
- code cleanup

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,12 +17,11 @@
 	"require": {
 		"php": "^7.1.3",
 		"ext-json": "*",
-		"illuminate/support": "~5.8.0|~5.9.0",
-		"illuminate/contracts": "~5.8.0|~5.9.0"
+		"illuminate/support": "~5.8.0"
```
Comment (austintoddj) [inline composer.json]: I'd like to keep ~5.9.0 in here. Removing contracts and keeping support is fine.

File: composer.json [hunk 2]
```diff
@@ -35,13 +34,10 @@
...
 		}
 	},
-	"scripts": {
-		"test": "vendor/bin/phpunit"
-	},
 	"extra": {
 		"laravel": {
 			"providers": [
-				"Canvas\\CanvasServiceProvider"
+				"Canvas\\ServiceProvider"
```
Comment (austintoddj) [inline composer.json]: I'd like to keep this as CanvasServiceProvider

File: readme.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ php artisan canvas:publish
 Run the tests with:
 
 ```bash
-composer test
+npm run test
```
Comment (austintoddj) [inline readme.md]: I'd like to keep this as `composer test`

File: resources/views/stats/index.blade.php [hunk 4]
```diff
@@ -19,7 +19,7 @@
                         <div class="card">
                             <div class="card-body">
                                 <h5 class="card-title text-muted small text-uppercase font-weight-bold">Views (30 days)</h5>
-                                <p class="card-text display-4">{{ \Canvas\SuffixedNumber::format($data['views']['count']) }}</p>
+                                <p class="card-text display-4">{{ $canvasNumberFormatter->format($data['views']['count']) }}</p>
```
Comment (austintoddj) [inline resources/views/stats/index.blade.php]: I'd like to keep this as it was, with SuffixedNumber as it's own class

File: resources/views/stats/show.blade.php [hunk 5]
```diff
@@ -42,7 +42,7 @@
                                 </p>
                             </div>
                             <div class="ml-auto">
-                                <span class="text-muted">{{ \Canvas\SuffixedNumber::format($views) }} View(s)</span>
+                                <span class="text-muted">{{ $canvasNumberFormatter->format($views) }} View(s)</span>
```
Comment (austintoddj) [inline resources/views/stats/show.blade.php]: I'd like to keep this as it was, with `SuffixedNumber` as it's own class

File: src/Console/InstallCommand.php [hunk 6]
```diff
@@ -75,7 +75,8 @@ private function registerCanvasServiceProvider()
 
         file_put_contents(config_path('app.php'), str_replace(
             "{$namespace}\\Providers\EventServiceProvider::class,".$eol,
-            "{$namespace}\\Providers\EventServiceProvider::class,".$eol."        {$namespace}\Providers\CanvasServiceProvider::class,".$eol,
+            "{$namespace}\\Providers\EventServiceProvider::class,".$eol
```
Comment (austintoddj) [inline src/Console/InstallCommand.php]: As long as this doesn't break the spacing on the output this is fine.

File: src/Events/PostViewed.php [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 namespace Canvas\Events;
 
-use Canvas\Post;
+use Canvas\Models\Post;
```
Comment (austintoddj) [inline src/Events/PostViewed.php]: I'd like to keep models in the `src` directory

File: src/Http/Controllers/MediaController.php [hunk 8]
```diff
@@ -14,11 +14,13 @@ class MediaController extends Controller
      */
     public function store(): string
     {
+        $diskName = config('canvas.storage_disk');
```
Comment (austintoddj) [inline src/Http/Controllers/MediaController.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Http/Controllers/PostController.php [hunk 9]
```diff
@@ -2,20 +2,23 @@
```
Comment (austintoddj) [inline src/Http/Controllers/PostController.php]: I'd like to keep this file as it was, as well as removing the IDE helpers. It's overkill to me.

File: src/Http/Controllers/StatsController.php [hunk 10]
```diff
@@ -2,8 +2,8 @@
```
Comment (austintoddj) [inline src/Http/Controllers/StatsController.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Http/Controllers/TagController.php [hunk 11]
```diff
@@ -2,17 +2,20 @@
```
Comment (austintoddj) [inline src/Http/Controllers/TagController.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Http/Controllers/TopicController.php [hunk 12]
```diff
@@ -2,17 +2,20 @@
```
Comment (austintoddj) [inline src/Http/Controllers/TopicController.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Listeners/StoreViewData.php [hunk 13]
```diff
@@ -2,7 +2,7 @@
 
 namespace Canvas\Listeners;
 
-use Canvas\Post;
+use Canvas\Models\Post;
```
Comment (austintoddj) [inline src/Listeners/StoreViewData.php]: I'd like to keep the models in the `src` directory

File: src/Models/Post.php [hunk 14]
```diff
@@ -1,18 +1,70 @@
 <?php
```
Comment (austintoddj) [inline src/Models/Post.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Models/Tag.php [hunk 15]
```diff
@@ -0,0 +1,87 @@
+<?php
```
Comment (austintoddj) [inline src/Models/Tag.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Models/Topic.php [hunk 16]
```diff
@@ -0,0 +1,96 @@
+<?php
```
Comment (austintoddj) [inline src/Models/Topic.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Models/View.php [hunk 17]
```diff
@@ -1,12 +1,38 @@
 <?php
```
Comment (austintoddj) [inline src/Models/View.php]: I'd like to keep this file as it was, as well as removing the IDE helper. It's overkill to me.

File: src/Models/View.php [hunk 18]
```diff
@@ -56,6 +82,7 @@ public static function viewTrend(Collection $views, int $days): array
 
         // Prep the array to perform a comparison with the actual view data
         $range = collect();
+        /** @var Carbon $date */
```
Comment (austintoddj) [inline src/Models/View.php]: I don't see a need for the comment here.

File: src/Relations/PostsTags.php [hunk 19]
```diff
@@ -1,10 +1,10 @@
 <?php
```
Comment (austintoddj) [inline src/Relations/PostsTags.php]: I'd like to keep the models in the `src` directory

File: src/Relations/PostsTopics.php [hunk 20]
```diff
@@ -1,10 +1,10 @@
 <?php
```
Comment (austintoddj) [inline src/Relations/PostsTopics.php]: I'd like to keep all models in the `src` directory

File: src/ServiceProvider.php [hunk 21]
```diff
@@ -2,15 +2,17 @@
```
Comment (austintoddj) [inline src/ServiceProvider.php]: I'd like to keep this file as it was before, with the name `CanvasServiceProvider`

File: src/Services/SuffixedNumberFormatter.php [hunk 22]
```diff
@@ -0,0 +1,72 @@
+<?php
```
Comment (austintoddj) [inline src/Services/SuffixedNumberFormatter.php]: I'd like to keep this class as it was before.

File: src/Traits/EventMap.php [hunk 23]
```diff
@@ -1,6 +1,9 @@
 <?php
```
Comment (austintoddj) [inline src/Traits/EventMap.php]: I'd like to keep the `EventMap` in the `src` directory

**Comment by reliq** [comment]
Closing in favour of https://github.com/cnvs/canvas/pull/462

**Pull request closed without merging**